scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,400 @@
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply brainmasks on maps
+    mask_image = nib.load(in_mask)
+    mask_data = mask_image.get_fdata(dtype=np.float32)
+    computed_map[np.where(mask_data == 0)] = 0
+
+    # Apply threshold based on combination of specific contrasts maps
+    for idx in idx_contrast_list:
+        computed_map[contrast_maps[idx] == 0] = 0
+
+    return computed_map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    # Select contrasts files from Bids folder
+    maps = []
+    layout = BIDSLayout(args.in_bids_folder, validate=False)
+    acquisition = layout.get_acquisition()
+
+    if args.resolution:
+        curr_res_acq = [res for res in acquisition if args.resolution[0] in res]
+        for idx, curr_acq in enumerate(curr_res_acq):
+            maps.append(layout.get(subject=args.in_participant,
+                                   datatype='anat', suffix='ihmt',
+                                   acquisition=str(curr_acq),
+                                   extension='nii.gz', return_type='file'))
+    else:
+        for curr_acq in acquisition:
+            maps.append(layout.get(subject=args.in_participant, datatype='anat',
+                                   suffix='ihmt', acquisition=str(curr_acq),
+                                   extension='nii.gz', return_type='file'))
+
+    # Set TR and FlipAngle parameters for ihMT and T1w images
+    parameters = []
+    parameters.append(set_acq_parameters(layout, maps[4][0]))
+    parameters.append(set_acq_parameters(layout, maps[5][0]))
+
+    # Fix issue from the presence of NaN into array
+    np.seterr(divide='ignore', invalid='ignore')
+
+    # Define reference image for saving ihMT files
+    ref_img = nib.load(maps[4][0])
+
+    # Define image output Name
+    if args.resolution:
+        out = [res for res in acquisition if args.resolution[0] in res]
+    else:
+        out = acquisition
+
+    # Create Coregistration images folers
+    if not os.path.isdir(os.path.join(args.out_dir, 'Coregistration_maps')):
+        os.mkdir(os.path.join(args.out_dir, 'Coregistration_maps'))
+
+    # Compute coregistration maps
+    warped_img = []
+    for idx, curr_acq in enumerate(maps):
+        if args.resolution:
+            warped_img.append(prepare_images(maps[idx], maps[4][0], args.out_dir,
+                                             out[idx].replace(args.resolution[0], '')))
+        else:
+            warped_img.append(prepare_images(maps[idx], maps[4][0], args.out_dir,
+                                             out[idx].replace(args.resolution[0], '')))
+
+    # Create contrasts maps folers
+    if not os.path.isdir(os.path.join(args.out_dir, 'Contrats_maps')):
+        os.mkdir(os.path.join(args.out_dir, 'Contrats_maps'))
```

Comment:
function exists assert_output_dirs_exist_and_empty

can be done one time at the very beginning of the script for all subjects all conditions...